DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/ViewDelegateController.cs

```diff
@@ -78,5 +82,52 @@ public IActionResult DeactivateDelegate(int delegateId)
 
             return RedirectToAction("Index", new { delegateId });
         }
+
+        [HttpGet]
+        [Route("{customisationId:int}/Remove")]
+        public IActionResult ConfirmRemoveFromCourse(int delegateId, int customisationId)
+        {
+            var centreId = User.GetCentreId();
+            var delegateUser = userDataService.GetDelegateUserCardById(delegateId);
+            var course = courseDataService.GetCourseNameAndApplication(customisationId);
+            if (delegateUser == null || delegateUser.CentreId != centreId || course == null)
+            {
+                return new NotFoundResult();
+            }
+
+            var model = new RemoveFromCourseViewModel
+            {
+                DelegateId = delegateUser.Id,
+                CustomisationId = customisationId,
+                CourseName = course.CourseName,
+                Name = delegateUser.FullName,
+                Confirm = false,
+            };
+            return View("ConfirmRemoveFromCourse", model);
+        }
+
+        [HttpPost]
+        [Route("{customisationId:int}/Remove")]
+        public IActionResult ExecuteRemoveFromCourse(int delegateId, int customisationId, RemoveFromCourseViewModel model)
+        {
+            if (!ModelState.IsValid)
+            {
+                return View("ConfirmRemoveFromCourse", model);
+            }
+
+            var centreId = User.GetCentreId();
+            var delegateUser = userDataService.GetDelegateUserCardById(delegateId);
+            if (delegateUser == null || delegateUser.CentreId != centreId ||

```

Comment:
Similarly the delegate user checks in this if statement can be removed thanks to the service filter.

Will want to double check whether the course service filter is appropriate too.